### PR TITLE
(test) O3-3853: Add Unit Tests for Datepicker Component

### PR DESCRIPTION
**Pull request by Samstar10**
# Requirements

- [x] This PR has a title that briefly describes the work done including the ticket number. Ensure your PR title includes a [conventional commit](https://o3-docs.openmrs.org/docs/frontend-modules/contributing#contributing-guidelines) label (such as `feat`, `fix`, or `chore`, among others). See existing PR titles for inspiration.

## For changes to apps

- [ ] My work conforms to the [**O3 Styleguide**](https://om.rs/styleguide) and [**design documentation**](https://om.rs/o3ui).

## If applicable

- [x] My work includes tests or is validated by existing tests.
- [ ] I have updated the [esm-framework mock](https://github.com/openmrs/openmrs-esm-core/blob/main/packages/framework/esm-framework/mock.tsx) to reflect any API changes I have made.

## Summary

This PR includes unit tests for the DatePicker component as mentioned in this ticket (https://openmrs.atlassian.net/issues/O3-3853).

### CODE REVIEW

File: packages/framework/esm-styleguide/src/datepicker/datepicker.test.tsx [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+    exists: jest.fn(),
+    loadNamespaces: jest.fn(),
+    loadLanguages: jest.fn(),
+    use: jest.fn(),
+    on: jest.fn(),
+    off: jest.fn(),
+    dir: jest.fn(),
+    cloneInstance: jest.fn(),
+  } as unknown as i18nType; // Casting the mock to the correct i18n type
+});
```
Comment (vasharma05) [inline packages/framework/esm-styleguide/src/datepicker/datepicker.test.tsx:25]:
```suggestion

	Object.defineProperty(window, 'i18next', {
		value: {
    language: 'en',
    changeLanguage: jest.fn(),
    t: jest.fn((key) => key), // Simple mock for translation function
    // Other properties and methods that might be required
    init: jest.fn(),
    getFixedT: jest.fn(),
    exists: jest.fn(),
    loadNamespaces: jest.fn(),
    loadLanguages: jest.fn(),
    use: jest.fn(),
    on: jest.fn(),
    off: jest.fn(),
    dir: jest.fn(),
    cloneInstance: jest.fn(),
  }
	})
```

**Review by vasharma05 — COMMENTED** [review]
Hi @Samstar10!
Can you please connect with Sachit and combine #1130 together.
Thanks!
CC: @NethmiRodrigo

File: packages/framework/esm-styleguide/src/datepicker/datepicker.test.tsx (hunk 1 above)
Comment (vasharma05) [reply packages/framework/esm-styleguide/src/datepicker/datepicker.test.tsx:25]: Refer: https://stackoverflow.com/questions/73964371/mock-window-location-in-typescript-jest-test

**Review by NethmiRodrigo — COMMENTED** [review]
Thanks for picking this up @Samstar10! Please see the suggestions and the build failures:

**Comment by brandones** [comment]
@Samstar10 are you still working on this? Is it clear to you how to move forward with it?

**Pull request closed without merging**